openhtf/io/output/upload_json.py

```diff
@@ -0,0 +1,81 @@
+"""Module for uploading JSON-formatted test record files to server."""
+
+import os
+import logging
+import requests
+
+from json import JSONEncoder
+from openhtf import conf
+from openhtf.io.output.json_factory import OutputToJSON
+
+conf.Declare('project', 'project name')
+conf.Declare('data_server', 'data server')
+conf.Declare('cm_server', 'cm server')
+_LOG = logging.getLogger(__name__)
+
+class UploadJSON(OutputToJSON):
+  """Return an output callback that upload JSON Test Records to data server.
+
+  An example filename_pattern might be:
+    '/data/test_records/%(dut_id)s.%(start_time_millis)s'
+
+  To use this output mechanism:
+    test = openhtf.Test(PhaseOne, PhaseTwo)
+    test.AddOutputCallback(UploadJson(
+        '/data/test_records/%(dut_id)s.%(start_time_millis)s'))
+  
+    in config.yaml:
+       data_server:  http://your_data_server_entry
+  Args:
+    filename_pattern: A format string specifying the filename to write to,
+      will be formatted with the Test Record as a dictionary.
+    inline_attachments: Whether attachments should be included inline in the
+      output.  Set to False if you expect to have large binary attachments.
+  """
+
+  def __init__(self, filename_pattern=None, inline_attachments=True, **kwargs):
+    super(UploadJSON, self).__init__(filename_pattern, inline_attachments, **kwargs)
+    self.headers = {'Accept' : 'application/json',
+                 'Content-Type' : 'application/json'}
+    self.data_server = None
+    self.project = None
+    self.cm_server = None
+
+
+  # pylint: disable=invalid-name
+  def __call__(self, test_record):
+    super(UploadJSON, self).__call__(test_record)
+  # pylint: enable=invalid-name
+    if not self.data_server:
+      self.project = conf.Config().project
```

Comment:
See the config fixes in that PR from last week. You should call Config() once then use the returned result to access keys rather than calling Config() every time you need to access a key.